etc/kubernetes/helm/enterprise-gateway/values.yaml

```diff
@@ -58,26 +103,26 @@ kernelspecsPvc:
 
 ingress:
   enabled: false
-
-  # Ingress resource host
-  hostName: ""
-
+  hostName: "" # Ingress resource host
+  pathType: "Prefix"
+  # for nginx:
+  # path: /gateway/?(.*)
+  # for traefik:
+  # path: "/gateway"
+  path: "/gateway"
+  annotations: {}
   # Ingress resource annotations to be included depending in ingress controller.
-  traefik:
-    enabled: true
-    path: /gateway
-    annotations:
-      kubernetes.io/ingress.class: "traefik"
-      traefik.frontend.rule.type: PathPrefixStrip
-
-  nginx:
-    enabled: false
-    path: /gateway/?(.*)
-    annotations:
-      kubernetes.io/ingress.class: "nginx"
-      nginx.ingress.kubernetes.io/rewrite-target: /$1
-      nginx.ingress.kubernetes.io/ssl-redirect: "false"
-      nginx.ingress.kubernetes.io/force-ssl-redirect: "false"
+  # traefik:
```

Comment:
Where are these annotations being set now? Does it become a user configuration step?

Reply:
yes, user will have to replace to set annotations as follow, see [doc](https://helm.sh/docs/chart_template_guide/values_files/):

> - A values file if passed into helm install or helm upgrade with the -f flag (helm install -f myvals.yaml ./mychart)
>- Individual parameters passed with --set (such as helm install --set foo=bar ./mychart)

with my changes user can set any annotations they want and use any ingress controller.

Reply:
@lresende any other questions?

Reply:
is there a way to enable users to have the following options: 
- nginx or default-nginx --> use the current annotations for nginx
- traefik or default-traefik --> use the current annotations for traefik
- custom -> user enters whatever they are comfortable with

Reply:
@lresende there is no issue with using nginx, traefik or any other ingress with the new configuration.
1.  Instead of one having hardcoded annotations for the ones you have mentioned, new configuration allows to set any annotation you want and use any ingress controller you want.
2. there are a lot of different ingress controllers that users can use, it will be a lot of overhead to keep track of all possible ingresses and corresponding annotations.
3. Kubernetes is deprecating use of [annotations](https://kubernetes.io/docs/concepts/services-networking/ingress/#deprecated-annotation) and is moving towards `IngressClass` resource instead.

I can do the following updates:
1. Update documentation on how to use new configuration and address nginx and traefik annotations as examples.
2. Add a warning that upgrade to new helm chart requires reviewing helm config.
3. add option to specify `ingressClassName` in Ingress resource

Reply:
My issue is how to simplify some of these for most used ingress (e.g. nginx and traefik) as we have done previously. Having said that, if it's a copy and paste from docs, then it seems like a good compromise.

Reply:
> if it's a copy and paste from docs or some other easy and visible place, then it seems like a good compromise.

Should we then replace the commented-out portion containing the old nginx and traefik settings with a commented-out link to the docs?  I believe this will make things less ambiguous to the user.